src/SnowflakeTransformation.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\SnowflakeTransformation;
+
+use Keboola\Component\UserException;
+use Keboola\SnowflakeDbAdapter\Connection;
+use Keboola\SnowflakeTransformation\Exception\DeadConnectionException;
+use Psr\Log\LoggerInterface;
+use Retry\BackOff\ExponentialBackOffPolicy;
+use Retry\Policy\SimpleRetryPolicy;
+use Retry\RetryProxy;
+
+class SnowflakeTransformation
+{
+    private Connection $connection;
+
+    private LoggerInterface $logger;
+
+    private array $databaseConfig;
+
+    public function __construct(Config $config, LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+        $this->databaseConfig = $config->getDatabaseConfig();
+        $this->connection = $this->createConnection();
+    }
+
+    public function setSession(Config $config): void
+    {
+        $sessionVariables = [];
+        $sessionVariables['QUERY_TAG'] = sprintf("'%s'", json_encode(['runId' => getenv('KBC_RUNID')]));
+
+        if ($config->getQueryTimeout()) {
+            $sessionVariables['STATEMENT_TIMEOUT_IN_SECONDS'] = $config->getQueryTimeout();
+        }
+
+        array_walk($sessionVariables, function (&$item, $key): void {
+            $item = vsprintf(
+                '%s=%s',
+                [
+                    $key,
+                    $item,
+                ]
+            );
+        });
+
+        $query = sprintf('ALTER SESSION SET %s;', implode(',', $sessionVariables));
+        $this->runRetryableQuery($query, 'alter session');
+    }
+
+    public function processSteps(array $steps): void
+    {
+        foreach ($steps as $step) {
+            foreach ($step['blocks'] as $block) {
+                $this->logger->info(sprintf('Processing block "%s".', $block['name']));
+                $this->executionQueries($block['name'], $block['script']);
+            }
+        }
+    }
+
+    public function executionQueries(string $name, array $queries): void
+    {
+        foreach ($queries as $query) {
+            $this->logger->info(sprintf('Running query "%s".', $query));
+            $this->runRetryableQuery($query, $name);
+        }
+    }
+
+    private function createConnection(): Connection
+    {
+        $connection = new Connection($this->databaseConfig);
+        return $connection;
+    }
+
+    private function runRetryableQuery(string $query, string $errorMessage): void
+    {
+        $retryPolicy = new SimpleRetryPolicy(
+            SimpleRetryPolicy::DEFAULT_MAX_ATTEMPTS,
+            ['PDOException', 'ErrorException']
```

Comment:
no tak ted uz to vubec nedava smysl podle me :) `query` pouziva ODBC, navic tam ma catch na throwable, takze ErrorException to taky nevyhodi

Reply:
nj máš pravdu :) tak jsem ho prozatím odstranil úplně